instrumentation-tests/src/androidTest/java/com/mapbox/navigation/instrumentation_tests/core/RouteRefreshOnDemandTest.kt

```diff
@@ -174,6 +178,193 @@ class RouteRefreshOnDemandTest : BaseTest<EmptyTestActivity>(EmptyTestActivity::
         )
     }
 
+    @Test
+    fun route_refresh_on_demand_after_alternatives_change_between_attempts() = sdkTest {
+        createMapboxNavigation(createRouteRefreshOptionsWithInvalidInterval(5_000))
+        mockWebServerRule.requestHandlers.add(
+            MockDirectionsRequestHandler(
+                "driving-traffic",
+                readRawFileText(activity, R.raw.route_response_route_refresh),
+                twoCoordinates
+            )
+        )
+        val refreshHandler = MockDirectionsRefreshHandler(
+            "route_response_route_refresh",
+            readRawFileText(activity, R.raw.route_response_route_refresh_annotations),
+        )
+        refreshHandler.jsonResponseModifier = DynamicResponseModifier()
+        mockWebServerRule.requestHandlers.add(NthAttemptHandler(refreshHandler, 1))
+        val observer = TestObserver()
+
+        mapboxNavigation.routeRefreshController.registerRouteRefreshStateObserver(observer)
+        mapboxNavigation.startTripSession()
```

Comment:
I didn't notice any location emulation, the NN will probably emulate movement among the route due to the lost signal. Is that expected?

Reply:
TODO: forgot to fix that.